### ISSUE TITLE
Precompile fail

### ISSUE THREAD

**Issue by davidanthoff** [Luthaf/Conda.jl#21]
I get an error with precompile that seems to be related to Conda:

````
julia> using ExcelReaders
INFO: Precompiling module ExcelReaders...
INFO: Recompiling stale cache file C:\Users\davidanthoff\.julia\lib\v0.4\ExcelReaders.ji for module ExcelReaders.
WARNING: Module Conda uuid did not match cache file
ERROR: __precompile__(true) but require failed to create a precompiled cache file
 in require at loading.jl:252
````

This happens on a fresh Win install, fresh julia 0.4 install, after I clone ``ExcelReaders``. Once I exit julia and do ``using ExcelReaders`` again everything works. Here is the full log:

````
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.0 (2015-10-08 06:20 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32

julia> Pkg.clone("ExcelReaders")
INFO: Initializing package repository C:\Users\davidanthoff\.julia\v0.4
INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl
INFO: Cloning ExcelReaders from git://github.com/davidanthoff/ExcelReaders.jl.git
INFO: Computing changes...
INFO: Cloning cache of ArrayViews from git://github.com/JuliaLang/ArrayViews.jl.git
INFO: Cloning cache of BinDeps from git://github.com/JuliaLang/BinDeps.jl.git
INFO: Cloning cache of Compat from git://github.com/JuliaLang/Compat.jl.git
INFO: Cloning cache of Conda from git://github.com/Luthaf/Conda.jl.git
INFO: Cloning cache of DataArrays from git://github.com/JuliaStats/DataArrays.jl.git
INFO: Cloning cache of DataFrames from git://github.com/JuliaStats/DataFrames.jl.git
INFO: Cloning cache of Dates from git://github.com/quinnj/Dates.jl.git
INFO: Cloning cache of Docile from git://github.com/MichaelHatherly/Docile.jl.git
INFO: Cloning cache of GZip from git://github.com/JuliaLang/GZip.jl.git
INFO: Cloning cache of JSON from git://github.com/JuliaLang/JSON.jl.git
INFO: Cloning cache of PyCall from git://github.com/stevengj/PyCall.jl.git
INFO: Cloning cache of Reexport from git://github.com/simonster/Reexport.jl.git
INFO: Cloning cache of SHA from git://github.com/staticfloat/SHA.jl.git
INFO: Cloning cache of SortingAlgorithms from git://github.com/JuliaLang/SortingAlgorithms.jl.git
INFO: Cloning cache of StatsBase from git://github.com/JuliaStats/StatsBase.jl.git
INFO: Cloning cache of StatsFuns from git://github.com/JuliaStats/StatsFuns.jl.git
INFO: Cloning cache of URIParser from git://github.com/JuliaWeb/URIParser.jl.git
INFO: Installing ArrayViews v0.6.4
INFO: Installing BinDeps v0.3.18
INFO: Installing Compat v0.7.6
INFO: Installing Conda v0.1.7
INFO: Installing DataArrays v0.2.19
INFO: Installing DataFrames v0.6.10
INFO: Installing Dates v0.4.4
INFO: Installing Docile v0.5.19
INFO: Installing GZip v0.2.18
INFO: Installing JSON v0.5.0
INFO: Installing PyCall v1.1.2
INFO: Installing Reexport v0.0.3
INFO: Installing SHA v0.1.2
INFO: Installing SortingAlgorithms v0.0.6
INFO: Installing StatsBase v0.7.4
INFO: Installing StatsFuns v0.1.4
INFO: Installing URIParser v0.1.1
INFO: Building PyCall
INFO: Precompiling module Compat...
INFO: Precompiling module JSON...
INFO: Precompiling module URIParser...
INFO: Precompiling module SHA...
INFO: PyCall is using python (Python 2.7.10) at C:\Python27\python.exe, libpython = python27
INFO: Package database updated

julia> using ExcelReaders
INFO: Precompiling module ExcelReaders...
INFO: Recompiling stale cache file C:\Users\davidanthoff\.julia\lib\v0.4\ExcelReaders.ji for module ExcelReaders.
WARNING: Module Conda uuid did not match cache file
ERROR: __precompile__(true) but require failed to create a precompiled cache file
 in require at loading.jl:252

julia> quit()
PS C:\Users\davidanthoff> .\AppData\Local\Julia-0.4.0\bin\julia.exe
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.0 (2015-10-08 06:20 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32

julia> using ExcelReaders

julia>
````

I'm CCing @stevengj, mainly because he seems to actually understand precompile really well. Feel free to ignore, not really your problem, although something similar might happen for your various packages like PyPlot (but I haven't tried).

**Comment by stevengj** [comment]
This is JuliaLang/julia#13458.  It is fixed in 0.5, but in 0.4 the workaround is just to restart Julia as you noticed.

**Comment by Luthaf** [comment]
Is there any thing we can do on the Conda.jl side for that?

**Comment by stevengj** [comment]
No.

**Comment by davidanthoff** [comment]
Oh well... I do think the julia 0.4 release was not well managed, though. This is just another example. Precompile was merged way too late in the release cycle, so that all the subsequent problems in the package world only show up now, after julia 0.4 was released. The experience of using julia 0.4 right now is just rough, with things like this and endless streams of depreciation warnings even from major packages. I hope with the next release there will be less large changes closer to release, more time to find issues like this before a release and more time for packages to make necessary adjustments.

Sorry for the rant, especially since no one in this thread is at fault for this!

**Comment by davidanthoff** [comment]
Actually, I'll leave this open until julia 0.5 is released, so that the next person running into this problem can actually see this discussion.

**Comment by mlubin** [comment]
Hopefully package installation being broken will be fixed in 0.4.1, not 0.5.

**Comment by Luthaf** [comment]
Now that 0.4.1 is out, can you confirm this is fixed ?

**Comment by stevengj** [comment]
It should be, since the [relevant patch](https://github.com/JuliaLang/julia/commit/d91a6d3410ee21af9ca56e55bfe4b4c40b7d46b8) seems to have been [merged in 0.4.1](https://github.com/JuliaLang/julia/commits/v0.4.1?page=2).

**Comment by davidanthoff** [comment]
Yes, I just tried, this is fixed. Thanks!
